### PR TITLE
[MINI-6258]Download Miniapp

### PR DESCRIPTION
**Pull request by rleojoseph**
# Description
* Download MiniApp using the new interface
* Also, check if the Miniapp has been downloaded properly.
* Updated Demo app to test both the features.

Demo : https://rak.box.com/s/lhsitbb0vyt9xlbf9o4wng5gktountsb

## Links
[MINI-6258](https://jira.rakuten-it.com/jira/browse/MINI-6258)

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [ ] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [ ] I removed all sensitive data before every commit, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by GondiTilak — APPROVED** [review]
Looks good to me :+1:

File: Sources/Classes/core/MiniApp.swift [hunk 1]
```diff
@@ -206,8 +206,33 @@ public class MiniApp: NSObject {
...
 
-    public static func unzipMiniApp(fileName: String, miniAppId: String, versionId: String) {
-        MiniAppSDKUtility.unzipMiniApp(fileName: fileName, miniAppId: miniAppId, versionId: versionId)
+    /// Unzip the provided archive file from bundle.
+    /// - Parameters:
+    ///   - fileName: File name of the Bundle that you want to extract in MiniApp folder
+    ///   - miniAppId: MiniApp ID
+    ///   - versionId: Version ID
+    ///   - completionHandler: Status of the Unzip feature
+    public static func unzipMiniApp(fileName: String, miniAppId: String, versionId: String, completionHandler: ((Result<Bool, MASDKError>) -> Void)? = nil) {
```
Comment (Climbatize) [inline Sources/Classes/core/MiniApp.swift:215]:
```suggestion
    public static func unzipMiniApp(fileName: String, miniAppId: String, versionId: String, completionHandler: @escaping ((Result<Bool, MASDKError>) -> Void)? = nil) {
```

File: Sources/Classes/core/MiniApp.swift [hunk 2]
```diff
@@ -206,8 +206,33 @@ public class MiniApp: NSObject {
...
+        realMiniApp.downloadMiniApp(appId: appId, versionId: versionId, completionHandler: completionHandler)
+    }
+
+    /// Check and return TRUE if the MiniApp is available for a given MiniAppID and VersionID.
+    /// This method not only checks if the folder is available, but also checks if index.html is available.
+    /// - Parameters:
+    ///   - appId: MiniApp ID
+    ///   - versionId: VersionID of the MiniApp
+    /// - Returns: TRUE if MiniApp is available
+    public static func isMiniAppDownloadedAlready(appId: String, versionId: String) -> Bool {
```
Comment (Climbatize) [inline Sources/Classes/core/MiniApp.swift:234]:
```suggestion
    public static func isMiniAppDownloaded(appId: String, versionId: String) -> Bool {
```

File: Example/Controllers/SwiftUI/Features/List/MiniAppListRowCell.swift [hunk 3]
```diff
@@ -50,17 +61,53 @@ struct MiniAppListRowCell: View {
...
+            case .success:
+                print("Download Completed")
+            case .failure(let error):
+                print("Error downloading Miniapp:", error)
+            }
+        }
+    }
+
+    func isMiniAppDownloadedAlready() {
+        if MiniApp.isMiniAppDownloadedAlready(appId: miniAppId, versionId: versionId) {
```
Comment (Climbatize) [inline Example/Controllers/SwiftUI/Features/List/MiniAppListRowCell.swift:93]:
```suggestion
        if MiniApp.isMiniAppDownloaded(appId: miniAppId, versionId: versionId) {
```

**Review by Climbatize — APPROVED** [review]

File: Sources/Classes/core/MiniApp.swift (hunk 1 above)
Comment (rleojoseph) [reply Sources/Classes/core/MiniApp.swift:215]: @Climbatize , closure is already optional. So we don't need to add escaping

**Review by Climbatize — APPROVED** [review]

**Pull request merged**